sdk/tests/features/feature_extractor.py

```diff
@@ -0,0 +1,52 @@
+import importlib
+import inspect
+import pkgutil
+
+from features.get_project_root import get_project_root
+
+
+def validate_feature_list(feature_list):
+    for feature in feature_list:
+        if feature == "":
+            raise ValueError("lusid_feature error: Some decorated methods have no value passed. "
+                             "Please make sure each lusid_feature decorator has a value code passed.")
+
+    unique_features = set(feature_list)
+    if len(unique_features) != len(feature_list):
+        sorted_features = sorted(feature_list)
+        for i, code in enumerate(sorted_features):
+            if sorted_features[i] == sorted_features[i + 1]:
+                raise ValueError(f'lusid_feature error: Feature code "{code}" is a duplicate. '
+                                 'Please make sure each feature code is unique.')
+
+
+def get_decorator_values_from_classes(module, feature_list):
+    classes = inspect.getmembers(module, predicate=inspect.isclass)
+    for cls_name, cls in classes:
+        methods = inspect.getmembers(cls)
+        for method_name, method in methods:  # Using method option as sourcecode parsing parses comments
+            if hasattr(method, "decorator_value"):
+                feature_list.append(method.decorator_value)
```

Comment:
change this to `yield method.decorator_value` and then add to the list where you extract, something like
```

def yield_func(source_list):
    for i in source_list:
        yield i * 2


orig_list = [1, 2, 3, 4, 5]

new_list = [i for i in yield_func(orig_list)]

print(new_list)
```